### DISCUSSION TITLE
Automatically upload dSYM files to Firebase from Codemagic

### DISCUSSION THREAD

**Discussion by FrazeColder** [codemagic-ci-cd/codemagic-docs#2648] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Hey there,

I am using Codemagic to build my Flutter app. Furthermore, I am using Firebase Crashlytics to keep track of my app. In order to properly use Firebase Crashlytics I need to upload the dSYM File of each build to Firebase. 

When I build my app locally I can use the following script according to Firebase Crashlytics:
```
/path/to/pods/directory/FirebaseCrashlytics/upload-symbols -gsp /path/to/GoogleService-Info.plist -p ios /path/to/dSYMs
```

In order to upload my dSYM file automatically I have added this script to Xcode: https://stackoverflow.com/a/64782202

However, I want Codemagic to build and upload my dYSM files when I build the app. Therefore, I need to follow this documentation and added this code to my `codemagic.yaml` file: https://docs.codemagic.io/knowledge-firebase/firebase-crashlytics-dsym-uploading/

Now I am facing multiple problems and questions:
1.	As you can see, I have multiple entries in my Crashlytics which have an Unknown Version. Where does it come from?
2.	Why do I have the version 89 multiple times with different UUIDs?
3.	Do I need the the script in Xcode when I build my App via Codemagic?
4.	How can I verify that Codemagic properly uploads the dSYM file? (as you can see currently there are just unknown versions with an uploaded dSYM file)
5.	Why are there some versions where the dSYM file is optional?

### Expected results

Having just one UUID per version with an uploaded dSYM from Codemagic and when I build my app locally to test and develope it, I want to have my dSYMs being uploaded automatically as well but from my local machine.

### Actual results

Mulitple and unknown versions without an uploaded dSYM

**Comment by dtrdic** [comment]
Hey @FrazeColder I can't see the content of your Firebase Console account, but I assume you're talking about the "Unknown version" issue
![image](https://github.com/user-attachments/assets/cc275c4b-9290-415a-b56f-ffe5624f9e65)

Please check [this SO discussion](https://stackoverflow.com/questions/77958081/uploaded-crashlytics-dsym-version-unknown) answer, which should explain the reason why is this happening

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
